### DISCUSSION TITLE
Detecting Harlem programatically

### DISCUSSION THREAD

**Discussion by Lexpeartha** [andrewcourtice/harlem#67] [General]
Hi, I'm considering adding a PR to vue-telescope-analyzer to support detecting harlem, so more people can discover the library if they see it is used on the website. I've tried checking the following, after having no success in rummaging through the code for any injectable/tracable property added to the vue: 
```
__VUE_DEVTOOLS_GLOBAL_HOOK__.apps[0].app.config.globalProperties 
```
however it seems that harlem doesn't inject something like `$harlem`, like the pinia does. So I'm wondering if there's a better way to approach detecting the library?

**Comment by Lexpeartha** [comment]
Hey @andrewcourtice 
Could you give me your opinion on this discussion?

**Comment by andrewcourtice** [comment]
Hi @Lexpeartha 

Thanks for taking the time to look into this. I will be releasing version 2.4 in the coming weeks which will include a global object you can use to detect Harlem with the Vue telemetry API.

The global object is likely to have a signature similar to this:
```typescript
window.$harlem.createInstance()
```
This may change so I will keep you updated.

Thanks 👍

**Comment by andrewcourtice** [comment]
@Lexpeartha The change you requested is now in. v3.0.0+ contains a `window.$harlem` object.
<img width="227" alt="image" src="https://user-images.githubusercontent.com/11718453/207998666-73fbfc58-972f-41d6-9b74-66e2208a24a7.png">

You should now be able to check for that property for Vue telemetry.
